### DISCUSSION TITLE
Add support for `SELECT ... FOR SHARE`

### DISCUSSION THREAD

**Discussion by ryanagillie** [planetscale/discussion#443] [Feature Requests] [answered]
I noticed `SELECT ... FOR UPDATE` was supported but NOT `SELECT ... FOR SHARE`. You can get around this by writing `SELECT ... LOCK IN SHARE MODE`, but the `SELECT ... FOR SHARE` syntax supports more features as noted in the [mysql documentation](https://dev.mysql.com/doc/refman/8.0/en/innodb-locking-reads.html)

**Comment by orware** [comment, marked as answer]
@rag4214,

Sorry for a delayed response to you on this one, but this sounds like it could be a good candidate for submission as an [issue over in the Vitess public repo](https://github.com/vitessio/vitess/issues) as a feature request as I don't believe there is an existing issue requesting this yet there.

Since the main feature discrepancies between PlanetScale databases and regular MySQL 8 instances are related to compatibility for those features not being built out just yet within our Vitess layer, making the request there will be helpful and if it is eventually added within Vitess we should get the support automatically within PlanetScale not long after.
